libcst/codemod/commands/convert_namedtuple_to_dataclass.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# pyre-strict
+from typing import List, Optional, Sequence
+
+import libcst as cst
+from libcst.codemod import VisitorBasedCodemodCommand
+from libcst.codemod.visitors import AddImportsVisitor, RemoveImportsVisitor
+from libcst.metadata import (
+    ProviderT,
+    QualifiedName,
+    QualifiedNameProvider,
+    QualifiedNameSource,
+)
+
+
+class ConvertNamedTupleToDataclassCommand(VisitorBasedCodemodCommand):
+    """
+    Convert NamedTuple class declarations to Python 3.7 dataclasses.
+
+    This only performs a conversion at the class declaration level.
+    It does not perform type annotation conversions, nor does it convert
+    NamedTuple-specific attributes and methods.
+    """
+
+    DESCRIPTION: str = "Convert legacy NamedTuple class declarations to Python 3.7 dataclasses."
+    METADATA_DEPENDENCIES: Sequence[ProviderT] = (QualifiedNameProvider,)
+
+    MODULE: str = "typing"
+    OBJECT: str = "NamedTuple"
+
+    def leave_ClassDef(
+        self, original_node: cst.ClassDef, updated_node: cst.ClassDef
+    ) -> cst.ClassDef:
+        new_bases: List[cst.Arg] = []
+        namedtuple_base: Optional[cst.Arg] = None
+        qualified_namedtuple: QualifiedName = QualifiedName(
+            name=f"{self.MODULE}.{self.OBJECT}", source=QualifiedNameSource.IMPORT
+        )
+
+        # Need to examine the original node's bases since they are directly tied to import metadata
+        for base_class in original_node.bases:
+            # Compare the base class's qualified named against the expected typing.NamedTuple
```

Comment:
```suggestion
            # Compare the base class' qualified name against the expected typing.NamedTuple
```

Reply:
@carljm  `class's` is legal for singular, which this is one at a time.  did you mean `classes'` to talk about the whole operation?

Reply:
@thatch Oops, yeah I was thinking of the plural. Made the comment due to the extraneous `d` and the other part was an afterthought that clearly needed a bit more thought :)